aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1938,7 +1938,7 @@ std::tuple<Tensor,Tensor> qr(const Tensor& self, bool some) {
     "should be replaced with\n",
     "Q, R = torch.linalg.qr(A, 'reduced' if some else 'complete')"
   );
-  std::string mode = some ? "reduced" : "complete";
+  c10::string_view mode = some ? "reduced" : "complete";
```

Comment:
Assigning computation to a non-owning object makes me very nervous haha. Good thing these are string constants. I'd feel better if you made this `const char* mode` instead.